codegen/files/service.go

```diff
@@ -1,46 +1,136 @@
+		// VarName is the service struct name.
+		VarName string
+		// Methods lists the service struct methods.
 		Methods []*serviceMethod
 	}
 
 	// serviceMethod describes a single service method.
 	serviceMethod struct {
 		// Name is the method name.
 		Name string
-		// Description is the method description.
-		Description string
-		// Payload is the payload type.
-		Payload *design.UserTypeExpr
-		// Result is the result type.
-		Result *design.UserTypeExpr
+		// Payload is the payload of the method.
+		Payload servicePayload
+		// HasPayload is true if the payload type is not empty.
+		HasPayload bool
+		// Result is the result of the method.
+		Result serviceResult
+		// HasResult is true if the result type is not empty.
+		HasResult bool
+	}
+
+	servicePayload struct {
+		// Name is the payload name.
+		Name string
+		// Fileds lists the payload fields.
+		Fields map[string]string
+	}
+
+	serviceResult struct {
+		// Name is the result name.
+		Name string
+	}
+
+	// serviceFile is the codgen file for a given service.
+	serviceFile struct {
+		service *design.ServiceExpr
 	}
 )
 
-// Service returns the file for the given service.
-func Service(api *design.APIExpr, service *design.ServiceExpr) codegen.File {
-	return nil
+// serviceTmpl is the template used to render the body of the service file.
+var serviceTmpl = template.Must(template.New("service").Parse(serviceT))
+
+// Service returns the service file for the given service.
+func Service(service *design.ServiceExpr) codegen.File {
+	return &serviceFile{service}
 }
 
-// serviceT is the template used to write a service definition.
-const serviceT = `
-type (
-	{{ comment .Description 1 }}
-	{{ .Name }} interface {{{ range .Methods }}
-		{{ comment .Description 2 }}
-		{{ .Name }}(context.Context, {{ gotypename .Payload }}) ({{ gotypename .Result }}, error)
+// Sections returns the service file sections.
+func (s *serviceFile) Sections(genPkg string) []*codegen.Section {
+	var (
+		data *serviceData
+	)
+	{
+		methods := make([]*serviceMethod, len(s.service.Endpoints))
+		for i, v := range s.service.Endpoints {
+			fields := make(map[string]string)
+			if o, ok := v.Payload.Attribute().Type.(design.Object); ok {
+				for key, value := range o {
+					fields[key] = codegen.GoTypeName(value.Type)
+				}
+			}
+			methods[i] = &serviceMethod{
+				Name: codegen.Goify(v.Name, true),
+				Payload: servicePayload{
+					Name:   codegen.Goify(v.Payload.Name(), true),
+					Fields: fields,
+				},
+				HasPayload: v.Payload != design.Empty,
+				Result: serviceResult{
+					Name: codegen.Goify(v.Result.Name(), true),
+				},
+				HasResult: v.Result != design.Empty,
+			}
+		}
+		data = &serviceData{
+			Name:    s.service.Name,
+			VarName: codegen.Goify(s.service.Name, true),
+			Methods: methods,
+		}
 	}
+
+	var (
+		header, body *codegen.Section
+	)
+	{
+		header = codegen.Header(s.service.Name+"Services", "services",
+			[]*codegen.ImportSpec{
+				&codegen.ImportSpec{Path: "context"},
```

Comment:
Thank you. https://github.com/goadesign/goa/pull/1139/commits/76a17d34a4cb7410b88d46b1aed22b1bf3f2e5b3